helix-rest/src/main/java/org/apache/helix/rest/server/HelixRestServer.java

```diff
@@ -162,6 +168,24 @@ protected ResourceConfig getResourceConfig(HelixRestNamespace namespace, Servlet
     return cfg;
   }
 
+  /*
+   * Initialize metric registry and jmx reporter for each namespace.
+   */
+  private void initMetricRegistry(ResourceConfig cfg, String namespace) {
```

Comment:
When this one is called? Do we need concurrency control?

Reply:
This will be called after a servlet is init. We don't need concurrency control same as `initServlet()`